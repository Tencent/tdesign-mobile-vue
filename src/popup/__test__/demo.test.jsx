/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/popup/demos/base.vue';
import customCloseVue from '@/popup/demos/custom-close.vue';
import mobileVue from '@/popup/demos/mobile.vue';
import withTitleVue from '@/popup/demos/with-title.vue';

const mapper = {
  baseVue,
  customCloseVue,
  mobileVue,
  withTitleVue,
};

describe('Popup', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Popup ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
