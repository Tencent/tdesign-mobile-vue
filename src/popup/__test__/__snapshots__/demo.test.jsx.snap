// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Popup > Popup baseVue demo works fine 1`] = `
<div
  class="popup-demo"
  data-v-70969ba8=""
>
  
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    data-v-70969ba8=""
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      顶部弹出
      
    </span>
    <!---->
  </button>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    data-v-70969ba8=""
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      左侧弹出
      
    </span>
    <!---->
  </button>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    data-v-70969ba8=""
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      中间弹出
      
    </span>
    <!---->
  </button>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    data-v-70969ba8=""
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      底部弹出
      
    </span>
    <!---->
  </button>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    data-v-70969ba8=""
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      右侧弹出
      
    </span>
    <!---->
  </button>
  
  <teleport-stub
    data-v-70969ba8=""
    disabled="false"
    to="[object HTMLBodyElement]"
  >
    <div
      class="t-overlay"
      style="z-index: 1000; transition-duration: 300ms; display: none;"
    >
      <!---->
    </div>
    <div
      class="t-popup t-popup--top"
      style="padding: 100px; display: none;"
    >
      <!--v-if-->
      
      
    </div>
  </teleport-stub>
</div>
`;

exports[`Popup > Popup customCloseVue demo works fine 1`] = `
<div
  class="popup-demo"
>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      居中弹出层-带自定义关闭按钮
      
    </span>
    <!---->
  </button>
  <teleport-stub
    disabled="false"
    to="[object HTMLBodyElement]"
  >
    <div
      class="t-overlay"
      style="z-index: 1000; transition-duration: 300ms; display: none;"
    >
      <!---->
    </div>
    <div
      class="t-popup t-popup--center"
      style="width: 240px; height: 240px; display: none;"
    >
      <!--v-if-->
      
      <svg
        class="t-icon t-icon-close-circle close-btn"
        color="#fff"
      >
        <use
          href="#t-icon-close-circle"
        />
      </svg>
      
    </div>
  </teleport-stub>
</div>
`;

exports[`Popup > Popup mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
  data-v-c2ceb286=""
>
  <h1
    class="title"
    data-v-c2ceb286=""
  >
    Popup 弹窗层
  </h1>
  <p
    class="summary"
    data-v-c2ceb286=""
  >
    由其他控件触发，屏幕滑出或弹出一块自定义内容区域
  </p>
  <div
    class="tdesign-mobile-demo-block"
    data-v-c2ceb286=""
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        01 组件类型
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        基础弹出层
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="popup-demo"
        data-v-70969ba8=""
        data-v-c2ceb286=""
      >
        
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          data-v-70969ba8=""
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            顶部弹出
            
          </span>
          <!---->
        </button>
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          data-v-70969ba8=""
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            左侧弹出
            
          </span>
          <!---->
        </button>
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          data-v-70969ba8=""
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            中间弹出
            
          </span>
          <!---->
        </button>
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          data-v-70969ba8=""
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            底部弹出
            
          </span>
          <!---->
        </button>
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          data-v-70969ba8=""
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            右侧弹出
            
          </span>
          <!---->
        </button>
        
        <teleport-stub
          data-v-70969ba8=""
          disabled="false"
          to="[object HTMLBodyElement]"
        >
          <div
            class="t-overlay"
            style="z-index: 1000; transition-duration: 300ms; display: none;"
          >
            <!---->
          </div>
          <div
            class="t-popup t-popup--top"
            style="padding: 100px; display: none;"
          >
            <!--v-if-->
            
            
          </div>
        </teleport-stub>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
    data-v-c2ceb286=""
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        02 组件示例
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        应用示例
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="popup-demo"
        data-v-099d70a8=""
        data-v-c2ceb286=""
      >
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          data-v-099d70a8=""
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            底部弹出层-带标题及操作
            
          </span>
          <!---->
        </button>
        <teleport-stub
          data-v-099d70a8=""
          disabled="false"
          to="[object HTMLBodyElement]"
        >
          <div
            class="t-overlay"
            style="z-index: 1000; transition-duration: 300ms; display: none;"
          >
            <!---->
          </div>
          <div
            class="t-popup t-popup--bottom"
            style="height: 258px; display: none;"
          >
            <!--v-if-->
            
            <div
              class="header"
              data-v-099d70a8=""
            >
              <div
                aria-role="button"
                class="btn btn--cancel"
                data-v-099d70a8=""
              >
                取消
              </div>
              <div
                class="title"
                data-v-099d70a8=""
              >
                标题文字
              </div>
              <div
                aria-role="button"
                class="btn btn--confirm"
                data-v-099d70a8=""
              >
                确定
              </div>
            </div>
            
          </div>
        </teleport-stub>
      </div>
      <div
        class="popup-demo"
        data-v-c2ceb286=""
      >
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            居中弹出层-带自定义关闭按钮
            
          </span>
          <!---->
        </button>
        <teleport-stub
          disabled="false"
          to="[object HTMLBodyElement]"
        >
          <div
            class="t-overlay"
            style="z-index: 1000; transition-duration: 300ms; display: none;"
          >
            <!---->
          </div>
          <div
            class="t-popup t-popup--center"
            style="width: 240px; height: 240px; display: none;"
          >
            <!--v-if-->
            
            <svg
              class="t-icon t-icon-close-circle close-btn"
              color="#fff"
            >
              <use
                href="#t-icon-close-circle"
              />
            </svg>
            
          </div>
        </teleport-stub>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`Popup > Popup withTitleVue demo works fine 1`] = `
<div
  class="popup-demo"
  data-v-099d70a8=""
>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    data-v-099d70a8=""
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      底部弹出层-带标题及操作
      
    </span>
    <!---->
  </button>
  <teleport-stub
    data-v-099d70a8=""
    disabled="false"
    to="[object HTMLBodyElement]"
  >
    <div
      class="t-overlay"
      style="z-index: 1000; transition-duration: 300ms; display: none;"
    >
      <!---->
    </div>
    <div
      class="t-popup t-popup--bottom"
      style="height: 258px; display: none;"
    >
      <!--v-if-->
      
      <div
        class="header"
        data-v-099d70a8=""
      >
        <div
          aria-role="button"
          class="btn btn--cancel"
          data-v-099d70a8=""
        >
          取消
        </div>
        <div
          class="title"
          data-v-099d70a8=""
        >
          标题文字
        </div>
        <div
          aria-role="button"
          class="btn btn--confirm"
          data-v-099d70a8=""
        >
          确定
        </div>
      </div>
      
    </div>
  </teleport-stub>
</div>
`;
